tests-functional/tests/test_positive_values_user_flow.py

```diff
@@ -0,0 +1,82 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as expected
+from selenium.webdriver.support.ui import WebDriverWait as Wait
+
+from pages.all_aboard import AllAboard
+from pages.firstrun import Firstrun
+from pages.sidebar import Sidebar
+
+
+@pytest.fixture
+def complete_firstrun(base_url, selenium):
+    """ Completes /firstrun interactions to start on-boarding"""
+    page = Firstrun(selenium, base_url).open()
+    page.complete_firstrun_values_flow()
+    return page
+
+
+@pytest.mark.nondestructive
+@pytest.mark.capabilities(firefox_args=['-foreground'])
+def test_successful_values_flow(base_url,
+                                selenium,
+                                complete_firstrun):
+    """Tests that a user can successfully complete the values track"""
+    # switch to browser chrome
+    selenium.set_context('chrome')
+
+    all_aboard = AllAboard(selenium)
+    sidebar = Sidebar(selenium)
+
+    # ensure that add-on action button is added as soon as we leave firstrun
+    Wait(selenium, 2).until(
+        expected.presence_of_element_located(
+            (By.ID, 'action-button--all-aboard-v1-all-aboard')
+        )
+    )
+
+    # enure that the sidebar is automatically shown on first notification
+    Wait(selenium, 3).until(
+        expected.presence_of_element_located((
+            By.CSS_SELECTOR,
+            '#content-deck #sidebar-box[sidebarcommand="viewWebPanelsSidebar"]'
+            )
+        )
+    )
+
+    previous_step = 0
+    while int(sidebar.current_step) < 5:
```

Comment:
There's a lot of logic going on in the test here, and you have selectors both in the test file and the page object. To make things easier to follow I'd suggest trying to move as much logic as possible to the page object. All you selectors could move there, and you could split up this logic into methods and properties that you could then use to assert things in the tests.

This is just a very high-level view, as I'm not that familiar with the code itself. Perhaps @davehunt might be able to chime in if he has time, but hopefully I'm not too far off the mark here on what he might suggest too.